### PR TITLE
[1/n] Add a dummy wheel test job to linux Github Action

### PR DESCRIPTION
**Pull request by tktrungna**
### Issue
Migrate tests depended on the build folder to a new job which runs tests from install folder (installed by the wheel package)

image

### Diff
* Adding a new wheel test job to Github Action for linux
* Adding a simple test script to be called by the new job
* Limit the change for linux-xenial-py3.6-gcc5.4 only
(next diff will start migrating tests from test job to wheel-test job)

### Test plan
check if all ci workflows pass

### CODE REVIEW

**Review by janeyx99 — COMMENTED** [review]
I don't think we should add this to PyTorch CI yet as it will use lots of resources to spin up and then wind down the machines to do a dummy test job. I think this can be the last step (versus the first). 

It may be good to keep this as a draft and come back to this when the wheel-test.sh script is written.

File: .github/templates/linux_ci_workflow.yml.j2 [hunk 1]
```diff
@@ -388,6 +388,97 @@ jobs:
...
+        run: |
+          docker pull "${DOCKER_IMAGE}"
+      - uses: seemethere/download-artifact-s3@0504774707cbc8603d7dca922e8026eb8bf3b47b
+        name: Download PyTorch Build Artifacts
+        with:
+          name: ${{ env.BUILD_ENVIRONMENT }}
+      - name: Unzip artifacts
+        run: |
+          # Unzip artifacts and remore build folder to fail any tests depending on it
+          unzip -o artifacts.zip && rm -rf build/
```
Comment (malfet) [inline .github/templates/linux_ci_workflow.yml.j2:426]: Can you just extract a `.whl` file from the artifacts folder and install it?

Comment (tktrungna) [reply .github/templates/linux_ci_workflow.yml.j2]: updated

**Comment by tktrungna** [comment]
> I don't think we should add this to PyTorch CI yet as it will use lots of resources to spin up and then wind down the machines to do a dummy test job. I think this can be the last step (versus the first).
> 
> It may be good to keep this as a draft and come back to this when the wheel-test.sh script is written.

On second thought, I have limited the change for linux-xenial-py3.6-gcc5.4 workflow only to avoid the impact on resources, while can keep the added job for my next PR depend on this one

**Pull request closed without merging**